demo/src/main/java/foundation/algorand/demo/provider/AVMProvider.kt

```diff
@@ -0,0 +1,109 @@
+package foundation.algorand.demo.provider
+
+import android.util.Log
+import com.algorand.algosdk.transaction.Transaction
+import com.algorand.algosdk.util.Encoder
+import com.fasterxml.uuid.Generators
+import foundation.algorand.crypto.EncoderType
+import foundation.algorand.crypto.avm.KeyPairs
+import foundation.algorand.provider.Message
+import foundation.algorand.provider.avm.models.*
+import java.security.KeyPair
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+
+
+/**
+ * A provider for the Algorand Virtual Machine (AVM).
+ * Used to test the provider.avm.models package.
+ */
+class AVMProvider(val providerId: String) {
```

Comment:
This class should be part of the provider package?

Reply:
In continuing with https://github.com/algorandfoundation/liquid-auth-android/pull/20#discussion_r1735353664, there are a few ways we could make this generic with a BaseProvider. It was included before we simplified, possibly it was over simplified?

I couldn't find a good shape that was useful outside of `Observables`, since we don't depend on the providers shape or interacting with multiple providers in a single application. It made sense to just make a concrete example of using the Models directly. 

```kotlin
enum class EncoderType {
    MSGPACK,
    BASE64,
    CBOR,
    NONE
}
type Message {
  payload: ByteArray
  encoding: EncoderType
}

interface IBaseProvider {
    fun handleMessage(message: Message): Any
}
```

Reply:
Lets to this! Thanks !